helpers/setupjsongenerator.py

```diff
@@ -0,0 +1,675 @@
+        :param backend_name: name of the backend in which to search
+        :type backend_name: str
+        :param osd_identifier: osd name to remove
+        :type osd_identifier: str
+        """
+        try:
+            for i in range(len(self.config['backends'])):
+                if self.config['backends'][i]['name'] == backend_name:
+                    self.config['backends'][i]['osds'].pop(osd_identifier)
+        except Exception:
+            pass
+
+    def add_vpool(self, storagerouter_ip, backend_name, preset_name, storage_ip, vpool_name=None):
+        """
+        Add a vpool to given storagerouter
+        :param storagerouter_ip
+        :type storagerouter_ip: str
+        :param vpool_name: name of the vpool to add
+        :type vpool_name: str
+        :param backend_name: name of the backend to link to the vpool
+        :type backend_name: str
+        :param preset_name: name of the preste to link to the vpool
+        :type preset_name: str
+        :param storage_ip:
+        :type storage_ip: str
+        """
+
+        if vpool_name is None:
+            vpool_name = 'myvpool{0}'.format(self.VPOOL_COUNTER)
+            SetupJsonGenerator.VPOOL_COUNTER += 1
+
+        required_params = {'storagerouter_ip': (str, Toolbox.regex_ip, True),
+                           'vpool_name': (str, None, False),
+                           'backend_name': (str, None, True),
+                           'preset_name': (str, None, True),
+                           'storage_ip': (str, Toolbox.regex_ip, True)}
+
+        actual_params = {'storagerouter_ip': storagerouter_ip,
+                         'vpool_name': vpool_name,
+                         'backend_name': backend_name,
+                         'preset_name': preset_name,
+                         'storage_ip': storage_ip}
+
+        Toolbox.verify_required_params(required_params=required_params, actual_params=actual_params, verify_keys=True)
+        self._valid_storagerouter(storagerouter_ip=storagerouter_ip)
+        self._validation_ip(ip=storage_ip)
+        if backend_name not in self._backends:
+            raise ValueError('Provided backend {0} not in known backends'.format(backend_name))
+        if preset_name not in self._presets:
+            raise ValueError('Provided preset not in known presets'.format(preset_name))
+        vpool_dict = {'backend_name': backend_name,
+                      'preset': preset_name,
+                      'storage_ip': storage_ip,
+                      'proxies': 1,
+                      'fragment_cache': {'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                                         'location': 'disk'
+                                         },
+                      'block_cache': {'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                                      'location': 'disk'
+                                      },
+                      'storagedriver': {'sco_size': 4,
```

Comment:
These should be configurable too

Reply:
What storagedriver params should be configurable exactly?